pkg/build/controller/build/build_controller.go

```diff
@@ -44,6 +45,53 @@ const (
 	maxRetries = 15
 )
 
+// resourceTriggerQueue tracks a set of resource keys to trigger when another object changes.
+type resourceTriggerQueue struct {
+	lock  sync.Mutex
+	queue map[string][]string
```

Comment:
Because it's not any more efficient for low cardinality builds, and string slice has lower memory footprint.  The queue already has a map and dedups multiples.